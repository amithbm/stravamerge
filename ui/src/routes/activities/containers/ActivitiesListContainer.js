import { connect } from 'react-redux'
import ActivitiesList from '../components/ActivitiesList'

const mapStateToProps = (state, props) => {
    const pageSize = props.pageSize
    const pageNumber = props.page
    const page = state.activities.summaries.slice(pageNumber * pageSize, pageNumber * pageSize + pageSize)

    return {
        activities: page,
        canMoveNext: (pageNumber+1) < state.activities.summaries.length/pageSize,
        canMovePrevious: pageNumber > 0,
        moveNext: props.moveNext,
        movePrevious: props.movePrevious
    }
}

const ActivitiesListContainer = connect(mapStateToProps)(ActivitiesList)

export default ActivitiesListContainer
