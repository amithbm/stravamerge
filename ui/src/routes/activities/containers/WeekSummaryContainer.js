import { connect } from 'react-redux'
import WeekSummary from '../components/WeekSummary'
import getWeekChartData from '../selectors/getWeekChartData.js'

const mapStateToProps = (state, props) => {
    return {
        weekData: getWeekChartData(state)
    }
}

const WeekSummaryContainer = connect(mapStateToProps)(WeekSummary)

export default WeekSummaryContainer
