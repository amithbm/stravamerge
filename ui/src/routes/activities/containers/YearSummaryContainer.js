import { connect } from 'react-redux'
import YearSummary from '../components/YearSummary'
import getYearChartData from '../selectors/getYearChartData'


const mapStateToProps = (state, props) => {
    return {
        yearData: getYearChartData(state)
    }
}

const YearSummaryContainer = connect(mapStateToProps)(YearSummary)

export default YearSummaryContainer