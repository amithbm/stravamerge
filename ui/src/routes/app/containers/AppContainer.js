import { connect } from 'react-redux'
import App from '../components/App'

const mapStateToProps = (state, props) => {
    return {
        isAuthenticated: state.auth.accessToken !== undefined
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        
    }
}

const AppContainer = connect(mapStateToProps, mapDispatchToProps)(App)

export default AppContainer