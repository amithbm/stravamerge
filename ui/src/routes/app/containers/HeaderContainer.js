import { connect } from 'react-redux'
import Header from '../components/Header'

const mapStateToProps = (state, props) => {
    return {
        isAuthenticated: state.auth.accessToken !== undefined
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        
    }
}

const HeaderContainer = connect(mapStateToProps, mapDispatchToProps)(Header)

export default HeaderContainer