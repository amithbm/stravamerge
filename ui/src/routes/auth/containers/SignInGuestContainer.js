import { connect } from 'react-redux'
import SignInGuest from '../components/SignInGuest'
import receiveTranslatedToken from '../../../state/auth/actions/receiveTranslatedToken'
import { withRouter } from 'react-router'

const mapStateToProps = (state, props) => {
    return {
        
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        signInAsGuest: (router) => {
            dispatch(receiveTranslatedToken('GUEST', 'Guest', 15753730))
            router.push('/dashboard')
        }
    }
}

const SignInGuestContainer = connect(mapStateToProps, mapDispatchToProps)(withRouter(SignInGuest))

export default SignInGuestContainer
