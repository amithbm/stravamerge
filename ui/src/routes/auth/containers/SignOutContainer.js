import { connect } from 'react-redux'
import SignOut from '../components/SignOut'
import { signOutLocal } from '../../../state/auth/actions/signOut'

const mapStateToProps = (state, props) => {
    return {
        
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        signOut: () => dispatch(signOutLocal())
    }
}

const SignOutContainer = connect(mapStateToProps, mapDispatchToProps)(SignOut)

export default SignOutContainer