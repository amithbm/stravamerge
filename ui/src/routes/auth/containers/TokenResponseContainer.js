import { connect } from 'react-redux'
import TokenResponse from '../components/TokenResponse'
import translateAccessToken from '../../../state/auth/actions/translateAccessToken'

const mapStateToProps = (state, props) => {
    return {
        
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        translateAccessToken: (token) => dispatch(translateAccessToken(token))
    }
}

const TokenResponseContainer = connect(mapStateToProps, mapDispatchToProps)(TokenResponse)

export default TokenResponseContainer