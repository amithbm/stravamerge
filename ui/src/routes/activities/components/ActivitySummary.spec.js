import React from 'react'
import renderer from 'react-test-renderer'
import { getActivity } from '../../../testing/testData'

describe('components', () => {
    describe('<ActivitySummary />', () => {
        jest.mock('../containers/YearSummaryContainer', () => () => 
            <div>Year Summary</div>
        )
        require('../containers/YearSummaryContainer')
        jest.mock('../containers/WeekSummaryContainer', () => () => 
            <div>Week Summary</div>
        )
        require('../containers/WeekSummaryContainer')

        const ActivitySummary = require('./ActivitySummary').default

        it('renders as expected', () => {
            const component = renderer.create(<ActivitySummary />)
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
