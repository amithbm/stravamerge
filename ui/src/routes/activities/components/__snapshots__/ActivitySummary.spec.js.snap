exports[`components <ActivitySummary /> renders as expected 1`] = `
<div>
  <h3
    className="m-t-0">
    History
  </h3>
  <div
    className="panel panel-default"
    id={undefined}>
    <div
      className="panel-body">
      <p
        className={undefined}>
        <span
          className="title-label">
          This Week
        </span>
      </p>
      <div>
        <div>
          Week Summary
        </div>
      </div>
      <p
        className="m-t-2">
        <span
          className="title-label">
          This Year
        </span>
      </p>
      <div>
        <div>
          Year Summary
        </div>
      </div>
    </div>
  </div>
</div>
`;
